packages/ember-data/lib/system/model/errors.js

```diff
@@ -297,6 +284,7 @@ export default Ember.Object.extend(Ember.Enumerable, Ember.Evented, {
 
     var content = get(this, 'content').rejectBy('attribute', attribute);
     get(this, 'content').setObjects(content);
+    get(this, 'errorsByAttributeName').remove(attribute);
```

Comment:
[Your're absolutely right](https://github.com/emberjs/ember.js/blob/v1.13.0/packages/ember-metal/lib/map.js#L356), nice catch! Will update to use `delete`...